include/tester/ssp/Softphone.hpp

```diff
@@ -1,15 +1,72 @@
 #pragma once
 
+#include<iostream>
+#include<string>
+#include<functional>
+#include<memory>
+
 #include"SSPAccount.hpp"
 #include"SSPCall.hpp"
+#include"SoftphoneArguments.hpp"
 
 class Softphone
 {
 public:
-    Softphone(/* args */);
-    ~Softphone();
+    Softphone(const SoftphoneArguments & args):
+        _account(args.id, args.domain, args.secret, std::bind(&Softphone::onIncomingCall, this, std::placeholders::_1))
+    {
+        _call = std::make_shared<SSPCall>(&_account, std::bind(&Softphone::onCallState, this,
```

Comment:
You can use this in the initializer list: `_call(std::make_shared<...>(...))`

Reply:
done